src/uu/top/src/tui.rs

```diff
@@ -0,0 +1,99 @@
+use crossterm::{
+    event::{self, Event, KeyCode},
+    terminal::{disable_raw_mode, enable_raw_mode},
+};
+use ratatui::{
+    backend::{Backend, CrosstermBackend},
+    layout::{Constraint, Direction, Layout},
+    style::{Color, Modifier, Style},
+    text::{Line, Text},
+    widgets::{Paragraph, Row, Table},
+    Terminal
+};
+use std::io::{Result, stdout};
+
+pub fn start_tui<F>(data_provider: F) -> Result<()>
+where
+    F: Fn() -> (Vec<String>, Vec<Vec<String>>),
+{
+    enable_raw_mode()?;
+    
+    let stdout = stdout();
+        
+    let backend = CrosstermBackend::new(stdout);
+    let mut terminal = Terminal::new(backend)?;
+
+    let result = run_app(&mut terminal, data_provider);
+
+    disable_raw_mode()?;
+    
+    terminal.show_cursor()?;
+
+    result
+}
+
+fn render_top_info() -> Vec<String> {
+    let info = "top - 04:30:13 up  6:02,  3 users,  load average: 0.40, 0.89, 1.24
+Tasks: 260 total,   3 running, 257 sleeping,   0 stopped,   0 zombie
+%Cpu(s):  1.9 us,  0.3 sy,  0.0 ni, 97.8 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
+MiB Mem :   3911.6 total,    190.6 free,   2549.1 used,   1171.9 buff/cache
+MiB Swap:   2048.0 total,    692.0 free,   1356.0 used.   1017.7 avail Mem";
+    
+    info.lines().map(String::from).collect()
+}
```

Comment:
For refreshing, I think it can be done asynchronously? Like refreshing every 3 seconds in the background and then reading the refresh here?

Reply:
And all data can be refreshed in background too, I think this is the easiest way to implement it?

Reply:
good idea! let me try this... will update the PR soon